### PR TITLE
NEXUS-24645: Asset event handlers to regenerate hosted metadata for Helm

### PR DESCRIPTION
**Pull request by aivasenko**
Ticket: https://issues.sonatype.org/browse/NEXUS-24645

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexFacetImpl.java [hunk 1]
```diff
@@ -88,6 +94,43 @@ public void on(RepositoryDeletedEvent deletedEvent) {
...
+
+  @Subscribe
+  @Guarded(by = STARTED)
+  @AllowConcurrentEvents
+  public void on(AssetPurgeEvent purged) {
+    log.debug("Updating index.yaml for hosted repository {}", getRepository().getName());
+    invalidateIndex();
+  }
+
+  private void maybeInvalidateIndex(final AssetEvent event) {
```
Comment (igorudovika) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexFacetImpl.java:129]: Not sure but seems it is needed here one more check. Like:
if (getRepository().getName() == event.getAsset().getRepository())
could you confirm that this event processing will not work for all Helm repositories?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexFacetImpl.java [hunk 2]
```diff
@@ -88,6 +94,43 @@ public void on(RepositoryDeletedEvent deletedEvent) {
     deleteIndexYaml();
   }
 
+  @Subscribe
+  @Guarded(by = STARTED)
+  @AllowConcurrentEvents
+  public void on(AssetCreateEvent created) {
+    log.debug("Updating index.yaml for hosted repository {}", getRepository().getName());
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/createindex/CreateIndexFacetImpl.java:101]: You can define a constant for `Updating index.yaml for hosted repository {}` statement

**Comment by Darkvar** [comment]
This PR depends on #156

**Comment by aivasenko** [comment]
Closed because all changes were cherry picked up by #156

**Pull request closed without merging**